internal/serialization/compact_serializer.go

```diff
@@ -53,11 +56,12 @@ func (CompactStreamSerializer) ID() int32 {
 }
 
 func (c CompactStreamSerializer) Read(input pubserialization.DataInput) interface{} {
-	schema := c.getOrReadSchema(input)
-	typeName := schema.TypeName()
-	serializer, ok := c.typeNameToSerializer[typeName]
+	// TODO: move context to the method signature
+	ctx := context.Background()
+	schema := c.getOrReadSchema(ctx, input)
```

Comment:
can schema be null?

Reply:
No, `getOrReadSchema` panics if ithe schema is not found.